### ISSUE TITLE
DynamicFont draw_char with outline can't draw the char's color

### ISSUE THREAD

**Issue by bengliezina** [godotengine/godot#64256]
### Godot version

v3.4.stable.mono.win64

### System information

Windows 10 GLES3

### Issue description

I tried to use `Control` and `DynamicFont` to draw texts on `Viewport` with C# script.Then I found that when I used `DynamicFont`'s method `DrawChar` without drawing outline,I could draw the char's color:
![image](https://user-images.githubusercontent.com/73519037/184058517-3ef85fa8-64f7-4f77-b918-fb1720b9db85.png)
but if I tried the draw the outline,it worked like:
![image](https://user-images.githubusercontent.com/73519037/184058621-6df4d227-ae1c-412e-a790-e65e087f6128.png)

when I changed the method into `Draw` to draw string,it can draw the outline and char's color in the meantime:
![image](https://user-images.githubusercontent.com/73519037/184058921-2f038809-56be-43ec-9282-3da0ce209547.png)

### Steps to reproduce

1. create `Control` node and `Font` node in `Viewport`
2. Load the font by using `DynamicFont`,and set the related setting such as outline size,outline color and font size
3. set `Font` 's value with `DynamicFont`
4. draw texts with using `Font` 's method `DrawChar`

### Minimal reproduction project

[TextTest.zip](https://github.com/godotengine/godot/files/9305850/TextTest.zip)

**Comment by timothyqiu** [comment]
This is expected behavior according to https://github.com/godotengine/godot/pull/17578#issuecomment-381097173

> it's a responsibility of clients to call draw_char twice - first to render an outline for the whole text and then to render the characters themselves.

So the `outline` parameter means "draw outline instead of the character itself". It's currently not documented.

**Comment by bengliezina** [comment]
> This is expected behavior according to [#17578 (comment)](https://github.com/godotengine/godot/pull/17578#issuecomment-381097173)
> 
> > it's a responsibility of clients to call draw_char twice - first to render an outline for the whole text and then to render the characters themselves.
> 
> So the `outline` parameter means "draw outline instead of the character itself". It's currently not documented.
> 
> p.s. `master` uses a separate `draw_char_outline()` instead of using the `outline` parameter.

Thanks for your reply. But I checked the C++ source code related to `DynamicFont`,and I got the `draw_char` method:
```
float DynamicFont::draw_char(RID p_canvas_item, const Point2 &p_pos, CharType p_char, CharType p_next, const Color &p_modulate, bool p_outline) const {
	if (!data_at_size.is_valid()) {
		return 0;
	}

	int spacing = spacing_char;
	if (p_char == ' ') {
		spacing += spacing_space;
	}

	if (p_outline) {
		if (outline_data_at_size.is_valid() && outline_cache_id.outline_size > 0) {
			outline_data_at_size->draw_char(p_canvas_item, p_pos, p_char, p_next, p_modulate * outline_color, fallback_outline_data_at_size, false, true); // Draw glyph outline.
		}
		return data_at_size->draw_char(p_canvas_item, p_pos, p_char, p_next, p_modulate, fallback_data_at_size, true, false) + spacing; // Return advance of the base glyph.
	} else {
		return data_at_size->draw_char(p_canvas_item, p_pos, p_char, p_next, p_modulate, fallback_data_at_size, false, false) + spacing; // Draw base glyph and return advance.
	}
}
```
Seems like it already do the two times draw when p_outline is true.It is in the `scene\resources\dynamic_font.cpp`,Godot version is v3.4.stable.mono.win64

**Comment by timothyqiu** [comment]
`DynamicFontAtSize::draw_char()` won't draw anything when the second-to-last parameter (`p_advance_only`) is true.

https://github.com/godotengine/godot/blob/850c5ccccff0b14b3de8a287bbc803a9523780b6/scene/resources/dynamic_font.h#L196

So the second `draw_char()` call only calculates the correct advance:

https://github.com/godotengine/godot/blob/850c5ccccff0b14b3de8a287bbc803a9523780b6/scene/resources/dynamic_font.cpp#L1216-L1223

**Comment by bengliezina** [comment]
Thanks for your reply,I completely understand now.

**Comment by akien-mga** [comment]
Fixed by #64297.
